builtins/amp-video.js

```diff
@@ -101,6 +124,80 @@ export function installVideo(win) {
     isVideoSupported_() {
       return !!this.video_.play;
     }
+
+    /**
+     * Safari on iOS does not trigger the `canplay` event for muted inline
+     * videos despite the fact that calls to `play` are allowed for muted inline
+     * videos in iOS 10.
+     * It does trigger `canplay` if `autoplay` attribute is set however.
+     * Since VideoManager handles autoplay behaviour for AMP, we do not
+     * propagate the `autoplay` attribute to the underlying video element but
+     * to get around the iOS bug, we will temporarily propagate `autoplay`
+     * attribute until we get the `canplay` event and then remove it.
+     * {@see https://bugs.webkit.org/show_bug.cgi?id=161804}
+     * @private
+     */
+    fixIOSCanplayEventForAutoplay_() {
+      if (!this.element.hasAttribute('autoplay') || !this.platform_.isIos()) {
+        return;
+      }
+
+      this.propagateAttributes(['autoplay'], this.video_);
+      listenOnce(this.video_, VideoEvents.CANPLAY, () => {
```

Comment:
nothing, we clean up after we get the CANPLAY. It is a `once` listener and then we remove the `autoplay` attribute we propagated, so we should be back to a clean state after canplay fires.